drivers: pwm: stm32: remove remaining Kconfig instances

gmarull opened the pull request:
Following other drivers, Kconfig based instances are now entirely removed. In order to do this change, PWM nodes in board DT files have been given a pwm{N} label so that both:

- DT API checks such as `#if DT_HAS_NODE(DT_NODELABEL(pwmN))` can be  used (`N` being PWM instance number).
- DT references can be written as `pwms = <&pwmN x y>;` instead of `pwms = <&{/soc/timers@XXXXXXXX/pwm} x y>;`

This approach is also used on the Linux Kernel. Reference: https://wiki.st.com/stm32mpu/wiki/TIM_device_tree_configuration#DT_configuration_examples

erwango commented on boards/arm/stm32f746g_disco/pinmux.c:
```diff
@@ -22,9 +22,9 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PC7, STM32F7_PINMUX_FUNC_PC7_USART6_RX},
 #endif
-#ifdef CONFIG_PWM_STM32_3
+#if DT_HAS_NODE(DT_NODELABEL(pwm2))
```
pwm3

erwango replied:
This being said, in device tree only pwm1 is enabled...

gmarull replied:
Fixed. Looks like pwm3 should be the enabled node. I can add a separate commit to fix that, even though it is out of the scope of this PR.

erwango reviewed (changes requested):
One comment, otherwise I'm fine with this proposal.
Also, it good to detail all the approaches in PR description, but maybe having them in commit message could bring confusion later on.

gmarull committed:
drivers: pwm: stm32: remove remaining Kconfig instances

Following other drivers, Kconfig based instances are now entirely
removed. In order to do this change, PWM nodes in board DT files have
been given a pwm{N} label so that both:

- DT API checks such as #if DT_HAS_NODE(DT_NODELABEL(pwmN)) can be
  used (N being PWM instance number).
- DT references can be written as pwms = <&pwmN x y>; instead of
  pwms = <&{/soc/timers@XXXXXXXX/pwm} x y>;

This approach is also used on the Linux Kernel.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull commented:
@erwango I have updated the PR with an approach that does not require SoC DTS changes and actually is the approach given by ST on Linux Kernel.

gmarull commented:
@henrikbrixandersen I have finally updated PR to follow the ST approach on Linux Kernel (see link in the description). I think it's the solution that for now has the smallest impact and at the same time follows the same convention as on Linux Kernel.

gmarull commented:
@erwango 4ce5ee1 is just a hack suggested by @carlescufi  to pass CI (it is timing out due to the large number of tests). It should be removed before merging.

erwango reviewed (approved):
Thanks for this change!

gmarull commented:
@erwango is it ok to remove CI fix now?

erwango commented:
> @erwango look at the STM32 board renames, I fixed a few, but many have the wrong string for the name in the board yaml file.

@galak will tell you

galak commented:
passed CI, removed the CI workaround.  Going to merge by hand.
